pkg/client/interceptor/intercept.go

```diff
@@ -0,0 +1,197 @@
+package interceptor
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type (
+	// GetFn is a function that can be used to intercept GET calls.
+	GetFn func(ctx context.Context, client client.WithWatch, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error
```

Comment:
Was it necessary to have the funcs as separate types? (vs just inline in the struct as before)

Reply:
No, I thought it was cleaner this way, but I removed it as it is not necessary